components/common/src/command/package/install.rs

```diff
@@ -467,51 +511,63 @@ impl<'a> InstallTask<'a> {
-                                &ident,
-                                self.channel,
-                                local.as_ref()
-                            ),
-                        )?;
-                        FullyQualifiedPackageIdent::from(local.as_ref().clone())?
-                    }
-                    Err(_) => {
-                        self.recommend_channels(ui, &ident, token)?;
-                        return Err(Error::PackageNotFound("If you are attempting to install from a local depot with an upstream configured, try again in a few seconds.".to_string()));
-                    }
-                },
+                Ok(latest_ident) => Some(latest_ident),
+                Err(Error::DepotClient(APIError(StatusCode::NotFound, _))) => None,
                 Err(e) => {
                     debug!("error fetching ident: {:?}", e);
                     return Err(e);
                 }
             };
 
-            // Return the latest identifier reported by the Builder API *unless* there is a newer
-            // version found installed locally.
-            match latest_local {
-                Ok(local) => {
-                    if local.as_ref() > latest_remote.as_ref() {
+            match (latest_local, latest_remote) {
+                (Ok(local), Some(remote)) => {
+                    if local.as_ref() > remote.as_ref() {
+                        // Return the latest identifier reported by
+                        // the Builder API *unless* there is a newer
+                        // version found installed locally.
                         ui.status(
                             Status::Found,
                             format!(
                                 "newer installed version ({}) than remote version ({})",
                                 &local,
-                                latest_remote.as_ref()
+                                remote.as_ref()
                             ),
                         )?;
                         Ok(local)
                     } else {
-                        Ok(latest_remote)
+                        Ok(remote)
+                    }
+                }
+                (Ok(local), None) => {
+                    if self.ignore_locally_installed_packages() {
+                        // This is the behavior that is currently feature-flagged
+                        self.recommend_channels(ui, &ident, token)?;
+                        ui.warn(format!(
+                            "Locally-installed package '{}' would satisfy '{}', \
+                             but we are ignoring that as directed",
+                            local.as_ref(),
+                            &ident,
+                        ))?;
+                        Err(Error::PackageNotFound("If you are attempting to install from a local depot with an upstream configured, try again in a few seconds.".to_string()))
```

Comment:
I think there's an error with `rustfmt` here. When I make changes to this line that should violate the standard (introducing random spaces) and are corrected elsewhere, `rustfmt` doesn't seem to notice.

I'd expect us to be consistent about whether long strings are split up and wrapped at column 100 like this:
```rust
                        Err(Error::PackageNotFound(
                            "If you are attempting to install from a local depot with an upstream \
                             configured, try again in a few seconds."
                                .to_string(),
                        ))
```